feat(traverse): add methods to `BoundIdentifier` + `MaybeBoundIdentifier` to create `SimpleAssignmentTarget`s

overlookmotel opened the pull request:
We had methods to create `AssignmentTarget`s. Add methods to create `SimpleAssignmentTarget`s too.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7418?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7011** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7011?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7122** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7122?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7418** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7418?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7418?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#7417** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7417?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Nov 22, 7:30 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
feat(traverse): add methods to `BoundIdentifier` + `MaybeBoundIdentifier` to create `SimpleAssignmentTarget`s (#7418)

We had methods to create `AssignmentTarget`s. Add methods to create `SimpleAssignmentTarget`s too.
